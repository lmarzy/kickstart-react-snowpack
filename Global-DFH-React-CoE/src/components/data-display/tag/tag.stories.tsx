import React from 'react';
import { Story, Meta } from '@storybook/react';

import { Tag } from './tag';

export default {
  title: 'Components/DataDisplay/Tag',
  component: Tag,
} as Meta;

export const Default: Story = ({ type, height, children }) => {
  return (
    <Tag type={type} height={height}>
      {children}
    </Tag>
  );
};

Default.args = {
  children: 'Tag Component',
};
