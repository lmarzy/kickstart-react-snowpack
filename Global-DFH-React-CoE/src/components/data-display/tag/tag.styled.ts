import styled, { css } from 'styled-components';

import { feedbackColors } from '../../..';
import { FeedbackType, HeightType } from '../../../shared/types-styles';
import { vr } from '../../../styles/utils/vr';

interface StyledSpanProps {
  $type: FeedbackType;
  $height: HeightType;
}

export const StyledSpan = styled.span<StyledSpanProps>`
  display: inline-flex;
  align-items: center;
  height: var(--height-${({ $height }) => $height});
  padding-left: var(--spacing-2);
  padding-right: var(--spacing-2);
  background-color: ${({ $type }) =>
    `var(--col-${feedbackColors($type).backgroundColor})`};
  border-width: 1px;
  border-style: solid;
  border-color: ${({ $type }) => `var(--col-${feedbackColors($type).borderColor})`};
  border-radius: 25px;

  ${({ $height }) =>
    $height === 'small' &&
    css`
      font-size: var(--size-1);
      line-height: ${vr(2)};
    `};
`;
