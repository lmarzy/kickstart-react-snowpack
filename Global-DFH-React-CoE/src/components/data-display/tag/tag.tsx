import React, { FC } from 'react';

import { FeedbackType, HeightType } from '../../../shared/types-styles';

import { StyledSpan } from './tag.styled';

interface TagProps {
  type: FeedbackType;
  height?: HeightType;
}

export const Tag: FC<TagProps> = ({ type, height = 'small', children }) => (
  <StyledSpan $type={type} $height={height}>
    {children}
  </StyledSpan>
);
