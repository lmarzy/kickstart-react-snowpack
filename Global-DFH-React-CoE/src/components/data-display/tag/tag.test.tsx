import React from 'react';
import { render, screen } from '@testing-library/react';

import { Tag } from './tag';

describe('Tag Component', () => {
  test('renders the info tag component', () => {
    render(<Tag type="info">Tag Component</Tag>);

    const tag = screen.getByText('Tag Component');

    expect(tag).toBeInTheDocument();
    expect(tag).toMatchSnapshot();
  });

  test('renders the success tag component', () => {
    render(<Tag type="success">Tag Component</Tag>);

    const tag = screen.getByText('Tag Component');

    expect(tag).toBeInTheDocument();
    expect(tag).toMatchSnapshot();
  });

  test('renders the warning tag component', () => {
    render(<Tag type="warning">Tag Component</Tag>);

    const tag = screen.getByText('Tag Component');

    expect(tag).toBeInTheDocument();
    expect(tag).toMatchSnapshot();
  });

  test('renders the error tag component', () => {
    render(<Tag type="error">Tag Component</Tag>);

    const tag = screen.getByText('Tag Component');

    expect(tag).toBeInTheDocument();
    expect(tag).toMatchSnapshot();
  });
});
