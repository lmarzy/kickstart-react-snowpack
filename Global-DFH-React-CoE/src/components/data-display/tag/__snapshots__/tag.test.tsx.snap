// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag Component renders the error tag component 1`] = `
<span
  class="sc-iJCRrE dHHpFQ"
>
  Tag Component
</span>
`;

exports[`Tag Component renders the info tag component 1`] = `
<span
  class="sc-iJCRrE FMsZk"
>
  Tag Component
</span>
`;

exports[`Tag Component renders the success tag component 1`] = `
<span
  class="sc-iJCRrE fuAHwE"
>
  Tag Component
</span>
`;

exports[`Tag Component renders the warning tag component 1`] = `
<span
  class="sc-iJCRrE fwXhLs"
>
  Tag Component
</span>
`;
